tools/timemachine/minifactory/statedb.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 IoTeX Foundation
```

Comment:
what's the problem of using the original statedb implementation with options?

Reply:
if use options, statedb struct will add member variable.  But the member variable is just used in tools, no matter with blockchain. So  inherit from statedb, just make changes in tools/timemachine.
Or  using ctx can be better?  like 
```
func (sdb *statedb) PutBlock() {
         ...
	if testCfg, ok := protocol.GetTestCtx(ctx); ok &&
		blk.Height() == testCfg.StopHeight && !testCfg.CommitBlock {
		return nil
	}
        ws.Commit(ctx)
        ...
}
```  
see the former commit:   [delete testCtx and realize in tools](https://github.com/iotexproject/iotex-core/pull/3692/commits/6756dc67d2a169d6515ffc08fc93cff3a5f8ce64)

Reply:
i think just use options, make a clear comment this flag/var is only used by timemachine tools

Reply:
use option to do and fixed